### ISSUE TITLE
Plugin causes issues with configuration cache

### ISSUE THREAD

**Issue by mikaoura** [craigatk/opentelemetry-gradle-plugin#187]
When adding the plugin to a project using gradle configuration cache I see the following errors:

Configuration cache problems found in this build.

2 problems were found storing the configuration cache.
- Plugin 'com.atkinsondev.opentelemetry-build': registration of listener on 'Gradle.addBuildListener' is unsupported
  See https://docs.gradle.org/8.4/userguide/configuration_cache.html#config_cache:requirements:build_listeners
- Plugin 'com.atkinsondev.opentelemetry-build': registration of listener on 'Gradle.addListener' is unsupported
  See https://docs.gradle.org/8.4/userguide/configuration_cache.html#config_cache:requirements:build_listeners
  
 (Noting this down to start with, haven't dug into causes or effects yet)

**Comment by craigatk** [comment]
Right on, that's one of the current limitations of the plugin https://github.com/craigatk/opentelemetry-gradle-plugin?tab=readme-ov-file#limitations

Gradle unfortunately doesn't have a drop-in replacement for the before/after listeners that works with the configuration cache. I'm exploring options to work with the Gradle build services instead to be compatible with the configuration cache, but it'll take a bit as it'll be a complete re-architecture of the plugin.

**Comment by craigatk** [comment]
I've updated the plugin at v2.2.0 to add experimental support for using the plugin with the configuration cache.

Docs are here, please give it a try and let me know how it works for you :) https://github.com/craigatk/opentelemetry-gradle-plugin?tab=readme-ov-file#gradle-configuration-cache

**Comment by mikaoura** [comment]
Thanks! We'll take a look.

**Comment by craigatk** [comment]
Thanks for the heads up about that error! I'll investigate 👀

**Comment by craigatk** [comment]
I released a new version `3.0.1` that should hopefully fix that issue. Please give it a try and let me know if it works for you :)

**Comment by mikaoura** [comment]
Right, the error is gone. However, the top-level span length is wrong when using configuration cache: it seems to count time from the time the configuration cache was created:

<img width="614" alt="image" src="https://github.com/user-attachments/assets/7d72694a-7c47-4fe1-bea2-2b3a6c6194ed" />

A colleague also pointed out, that with previous versions no telemetry is produced when configuration cache is used.

**Comment by craigatk** [comment]
Thanks for trying it out! I'll look into what's going on with the duration of the top-level span 👀

**Comment by craigatk** [comment]
I released a new version of the plugin `3.1.1` that will hopefully fix that start-time problem with the root span. Please give it a try and let me know if it works for you :)

**Comment by mikaoura** [comment]
Thanks a lot!

Looks promising after testing a couple of cases. Will close the issue once we have a few days of data (prob. after the holidays).

**Comment by craigatk** [comment]
Sounds good, thanks for testing it out!
